Common/GPU/OpenGL/GLRenderManager.cpp

```diff
@@ -200,6 +200,7 @@ bool GLRenderManager::ThreadFrame() {
 	std::unique_lock<std::mutex> lock(mutex_);
 	if (!run_)
 		return false;
+	bool vrlock = false;
```

Comment:
I don't really understand why this flag fixes a problem. 

If the issue is that PreVRRender gets called multiple times in a frame, can you just lift it out of the do {} loop below?

Reply:
`PreVRRender()` updates timestamps and the head pose. If I move it before the `do` loop then OpenXR calculates the waiting loop (inside of `do` loop) into rendering time. It ends with very shuttery rendering.

Reply:
Okay. I'll merge this if you add a clear comment explaining it, because it's really not very obvious when just looking at the code.

Reply:
done